### PR TITLE
API: enable access to user pictures and inline images

### PR DESCRIPTION
**Pull request by AdrienClairembault**
* Add a new API endpoint `userPicture`.

* Allow inline images to be accessed using an API token.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: inc/apirest.class.php [hunk 1]
```diff
@@ -215,6 +215,8 @@ public function call() {
          }
          return $this->returnResponse($this->lostPassword($this->parameters));
 
+      } else if (preg_match('%[Uu]ser/(\d+)/[Pp]icture%', $path_info, $matches)) {
```
Comment (cedric-anne) [inline inc/apirest.class.php]:
```suggestion
      } else if (preg_match('%user/(\d+)/picture%i', $path_info, $matches)) {
```

File: tests/web/APIRest.php [hunk 2]
```diff
@@ -110,14 +116,21 @@ protected function query($resource = "", $params = [], $expected_code = 200, $ex
...
-      $body            = $res->getBody();
-      $data            = json_decode($body, true);
-      if (is_array($data)) {
-         $data['headers'] = $res->getHeaders();
+
+      // retrieve data
+      $body = $res->getBody();
+
+      if ($file) {
+         $data = md5($body);
```
Comment (cedric-anne) [inline tests/web/APIRest.php]: Why md5 ?

File: inc/apirest.class.php (hunk 1 above)
Comment (AdrienClairembault) [reply inc/apirest.class.php]: Done in a46b369.

File: tests/web/APIRest.php (hunk 2 above)
Comment (AdrienClairembault) [reply tests/web/APIRest.php]: Done in 2cb093b.

**Pull request merged**